ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V122__participant_metering.sql

```diff
@@ -0,0 +1,32 @@
+
+--  Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+-- Transaction metering alterations
+SELECT 'Harmonise the transaction_metering columns with other examples in the db';
+
+ALTER TABLE transaction_metering ALTER application_id TYPE TEXT;
+ALTER TABLE transaction_metering ALTER ledger_offset TYPE TEXT;
+
+-- Parameter alterations
+SELECT 'Add additional ledger metering columns to parameters';
+
+ALTER TABLE parameters ADD COLUMN ledger_metering_end TEXT;
```

Comment:
parameters table is very special in the indexdb. It contains a single row that is changing very rapidly. I have seen this table occupy a lot of disk space though due to the fact of how PG is dealing with versions of the records. 
I am afraid that by adding two fields to this table (one of them quite big) we are doing ourselves a disservice. We are asking for more space, we are prolonging time critical ledger end updates and we are sending a curved ball to the postgres/oracle query planner.
I would consider creating a dedicated table just for these two columns

Reply:
I completely agree, also from separation of concerns point of view it would be much more ideal: thinking about removing metering altogether from participant db, having it in a separate namespace etc. Also it would be best to keep parameters table code (ParametersStorageBackend) and friends decoupled from metering